internal/commands/app/create_test.go

```diff
@@ -0,0 +1,191 @@
+			createdApp = realm.App{
+				GroupID: groupID,
+				Name:    name,
+				AppMeta: meta,
+			}
+			return createdApp, nil
+		}
+		client.ImportFn = func(groupID, appID string, appData interface{}) error {
+			return nil
+		}
+
+		cmd := &CommandCreate{
+			inputs: createInputs{newAppInputs: newAppInputs{
+				Name:            "test-app",
+				Project:         "test-project",
+				Location:        realm.LocationVirginia,
+				DeploymentModel: realm.DeploymentModelGlobal,
+			}},
+			realmClient: client,
+		}
+
+		assert.Nil(t, cmd.Handler(profile, nil))
+
+		data, readErr := ioutil.ReadFile(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name, local.FileRealmConfig.String()))
+		assert.Nil(t, readErr)
+
+		var config local.AppRealmConfigJSON
+		assert.Nil(t, json.Unmarshal(data, &config))
+		assert.Equal(t, local.AppRealmConfigJSON{local.AppDataV2{local.AppStructureV2{
+			ConfigVersion:   realm.DefaultAppConfigVersion,
+			Name:            "test-app",
+			Location:        realm.LocationVirginia,
+			DeploymentModel: realm.DeploymentModelGlobal,
+		}}}, config)
+		assert.Equal(t, realm.App{
+			GroupID: "test-project",
+			Name:    "test-app",
+			AppMeta: realm.AppMeta{
+				Location:        realm.LocationVirginia,
+				DeploymentModel: realm.DeploymentModelGlobal,
+			},
+		}, createdApp)
+	})
+
+	t.Run("should create a templated app when from is set", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+
+		testApp := realm.App{
+			ID:          primitive.NewObjectID().Hex(),
+			GroupID:     primitive.NewObjectID().Hex(),
+			ClientAppID: "from-app-abcde",
+			Name:        "from-app",
+		}
+
+		var zipCloser *zip.ReadCloser
+		defer func() { zipCloser.Close() }()
+
+		var createdApp realm.App
+		client := mock.RealmClient{}
+		client.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{testApp}, nil
+		}
+		client.ExportFn = func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
+			zipPkg, err := zip.OpenReader("testdata/project.zip")
+
+			zipCloser = zipPkg
+			return "", &zipPkg.Reader, err
+		}
+		client.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+			createdApp = realm.App{
+				GroupID: groupID,
+				Name:    name,
+				AppMeta: meta,
+			}
+			return createdApp, nil
+		}
+		client.ImportFn = func(groupID, appID string, appData interface{}) error {
+			return nil
+		}
+
+		cmd := &CommandCreate{
+			inputs: createInputs{newAppInputs: newAppInputs{
+				From:            testApp.Name,
+				Project:         testApp.GroupID,
+				Location:        realm.LocationIreland,
+				DeploymentModel: realm.DeploymentModelGlobal,
+			}},
+			realmClient: client,
+		}
+
+		assert.Nil(t, cmd.Handler(profile, nil))
+
+		data, readErr := ioutil.ReadFile(filepath.Join(profile.WorkingDirectory, cmd.inputs.From, local.FileRealmConfig.String()))
```

Comment:
could we use `local.LoadApp` here instead?  and the ensuing assert could just be `assert.Equal(t, local.AppRealmConfigJSON{...}, localApp.AppData)`